### ISSUE TITLE
In node, lua `require 'file'`, `lual_dofile`, `luaL_loadfile`, etc. doesn't seem to be working with non-empty files

### ISSUE THREAD

**Issue by cableray** [fengari-lua/fengari#182]
So I have a lua runtime that can run `luaL_dostring` calls just fine. But when I call `require 'other_lua_file_in_load_path'` in `luaL_dostring`, I get the error:
```
(2): error loading module 'other_lua_file_in_load_path' from file './other_lua_file_in_load_path.lua':
	(null)
```

Similarly, loading the file with `luaL_dofile`, `luaL_loadfile`, or `luaL_loadfilex` (mode `'t'`) all return an error code 2 and a null error message on the stack. This doesn't appear to be a syntax error in the file, the `lua` cli runs it just fine, and the file is deliberately simple. Additionally, if the file is _completely_ empty (no newlines or anything), then the file loads fine. (but returns nothing, of course)

I'm using `node v13.11.0`, `fengari 0.1.4`, and `fengari-interop 0.1.2` on OS X 10.15.5

How do I proceed in debugging this?

**Comment by cableray** [comment]
Ok, so I was testing this in a jest environment. It was having issues there, but running it directly seems to be working... That is odd...

**Comment by cableray** [comment]
Yes, confirmed that in jest tests, the code does not seem able to load files, but outside jest tests, the code works fine.  So I assume there's some global state or the like that jest is messing with, but I'm not sure what that might be.

**Comment by daurnimator** [comment]
fengari itself uses jest; so this seems odd...
Could you share your tests?
And please share the error message left on the stack.

**Comment by cableray** [comment]
I added the details and an example project, hope that helps. There doesn't seem to be anything useful on the stack. I ran a debugger, and was looking at the buffers lua was parsing as it was loading, and they didn't seem to match my file contents. that may have been because they weren't file buffers, and I was confused though. but they were full of non-printable characters (maybe \0? I didn't check though). Hopefully you get the same result that I did when running the example. If that's not the case, then I'll see about debugging more.

**Comment by fmhdp** [comment]
I just stumbled upon the same problem. For me it was require in lua failing that made me break out the debugger.
In the end the problem for me was a failing type check and the solution was setting the tes environment in Jest.

I've written more details on an issue filed with Jest here:
https://github.com/facebook/jest/issues/10368#issuecomment-829150932

**Comment by daurnimator** [comment]
Thanks for digging a bit @fmhdp . This sounds like a jest issue.

However, it might be possible to rewrite https://github.com/fengari-lua/fengari/blob/master/src/lauxlib.js#L903-L944 to not use `Buffer` at all and tiptoe around this issue. If you want to put in the research and work, I'm happy to review+merge it.

**Comment by fmhdp** [comment]
It's good to know that fengari is still maintained and contributions are welcome. I will keep that in mind :)

Since this problem vanishes when setting the correct test environment with Jest, I'm not sure if rewriting this part is really necessary. If I can find some time for open source contributions to fengari, I'd probably focus on the io library instead, since I could use support for writing files to disk in a node environment.
